src/tls.py

```diff
@@ -144,19 +144,16 @@ def _trusted_relation_created(self, _) -> None:
 
     def _trusted_relation_joined(self, event: RelationJoinedEvent) -> None:
         """Generate a CSR so the tls-certificates operator works as expected."""
-        if not event.app:
-            return
-
-        if not self.private_key or self.keystore_password or self.truststore_password:
+        if not self.enabled:
             logger.debug("Missing TLS relation, deferring")
             event.defer()
             return
 
-        alias = self.generate_alias(app_name=event.app.name, relation_id=event.relation.id)
+        alias = self.generate_alias(app_name=event.app.name, relation_id=event.relation.id)  # type: ignore
```

Comment:
it is better to be more specific when adding ignore for linters
```suggestion
        alias = self.generate_alias(app_name=event.app.name, relation_id=event.relation.id)  # pyright: ignore[reportOptionalMemberAccess]
```

Reply:
Agree with Mykola. I prefer not to ignore, as we're working with the typing limitations of our library. But it's unlikely to change, so let's be specific in what we're ignoring :)